### PR TITLE
Improve the error message for 'ballerina pull' command

### LINKED ISSUE #18466: Error on invalid `ballerina pull` module could improve
**Description:**
$title.

**Steps to reproduce:**
```cmd
$ ballerina pull twitter
error: no module-name provided
```

The issue here is that the org name is missing. Would be better if we could say what the expected format is.

**Affected Versions:**
```cmd
$ ballerina -v
Ballerina 1.0.0-beta
Language specification 2019R3
Ballerina tool 1.0.0
```

### PR DESCRIPTION
**Pull request by suganyasuven**
Fixes #18466

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/cmd/PullCommand.java [hunk 1]
```diff
@@ -81,28 +81,34 @@ public void execute() {
...
+            orgName = module[0];
+            packageName = module[1];
             if (orgName.equals("ballerina")) {
                 throw LauncherUtils.createLauncherException("`Ballerina` is the builtin organization and its modules"
                                                                     + " are included in the runtime.");
             }
         } else {
-            throw LauncherUtils.createLauncherException("no module-name provided");
+            CommandUtil.printError(outStream,
+                    "Pull command requires the names of the organization and module.",
```
Comment (pramodya1994) [inline cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/cmd/PullCommand.java]:
```suggestion
                    "pull command requires the names of the organization and module.",
```
Aren't we using simple letters only for warnings?

File: cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/cmd/PullCommand.java [hunk 2]
```diff
@@ -147,4 +155,12 @@ public void printUsage(StringBuilder out) {
     @Override
     public void setParentCmdParser(CommandLine parentCmdParser) {
     }
+
+    private String getPullCommandRegex() {
+        return MODULE_NAME_REGEX;
+    }
+
+    public boolean validateModuleName(String str) {
+        return (Pattern.matches(getPullCommandRegex(), str));
```
Comment (warunalakshitha) [inline cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/cmd/PullCommand.java]:
```suggestion
        return Pattern.matches(getPullCommandRegex(), str);
```

File: cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/cmd/PullCommand.java [hunk 3]
```diff
@@ -81,28 +83,34 @@ public void execute() {
...
+            throw LauncherUtils.createLauncherException("`Ballerina` is the builtin organization and its modules"
+                    + " are included in the runtime.");
         }
 
         // Get module name
-        int packageNameIndex = resourceName.indexOf(":");
-        if (packageNameIndex != -1) { // version is provided
-            moduleName = resourceName.substring(orgNameIndex + 1, packageNameIndex);
-            version = resourceName.substring(packageNameIndex + 1, resourceName.length());
+        if (packageName.contains(":")) { // version is provided
```
Comment (warunalakshitha) [inline cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/cmd/PullCommand.java]: Shall we use one spilt function and use return array to get module name and version? we can remove contains and check split array

File: cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/cmd/PullCommand.java [hunk 4]
```diff
@@ -81,28 +83,34 @@ public void execute() {
...
-            orgName = resourceName.substring(0, orgNameIndex);
-            if (orgName.equals("ballerina")) {
-                throw LauncherUtils.createLauncherException("`Ballerina` is the builtin organization and its modules"
-                                                                    + " are included in the runtime.");
-            }
-        } else {
-            throw LauncherUtils.createLauncherException("no module-name provided");
+        if (!validateModuleName(resourceName)) {
+            CommandUtil.printError(outStream,
+                    "invalid module name. Provide the module-name with the org-name",
```
Comment (warunalakshitha) [inline cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/cmd/PullCommand.java]: space after org name

File: cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/cmd/PullCommand.java [hunk 5]
```diff
@@ -81,28 +83,34 @@ public void execute() {
...
+                    false);
+            Runtime.getRuntime().exit(1);
+            return;
+        }
+
+        String[] module = resourceName.split("/");
+        orgName = module[0];
+        packageName = module[1];
+        if (orgName.equals("ballerina")) {
+            throw LauncherUtils.createLauncherException("`Ballerina` is the builtin organization and its modules"
```
Comment (warunalakshitha) [inline cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/cmd/PullCommand.java]: start with simple letter and shall we remove starting space.

File: cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/cmd/Constants.java [hunk 6]
```diff
@@ -37,4 +37,9 @@
     static final String SEARCH_COMMAND = "search";
     static final String CLEAN_COMMAND = "clean";
     static final String UNINSTALL_COMMAND = "uninstall";
+
+    static final String MODULE_NAME_REGEX = "[^0-9_][_\\w]+/[^0-9_][_\\.\\w]+|" +
```
Comment (warunalakshitha) [inline cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/cmd/Constants.java]: Shall we add a comment on describing regex rules.

File: cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/cmd/PullCommand.java (hunk 5 above)
Comment (warunalakshitha) [reply cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/cmd/PullCommand.java]:
```suggestion
            throw LauncherUtils.createLauncherException("`ballerina` is the builtin organization and its modules"
```

File: cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/cmd/PullCommand.java [hunk 7]
```diff
@@ -81,28 +83,35 @@ public void execute() {
 
         String resourceName = argList.get(0);
         String orgName;
+        String packageName;
         String moduleName;
         String version;
 
         // Get org-name
```
Comment (praneesha) [inline cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/cmd/PullCommand.java]:
```suggestion
        // Get the name of the organization.
```

File: cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/cmd/PullCommand.java [hunk 8]
```diff
@@ -81,28 +83,35 @@ public void execute() {
...
+
+        String[] moduleInfo = resourceName.split("/");
+        orgName = moduleInfo[0];
+        packageName = moduleInfo[1];
+        if (orgName.equals("ballerina")) {
+            throw LauncherUtils.createLauncherException("`ballerina` is the builtin organization and its modules"
+                    + " are included in the runtime.");
         }
 
         // Get module name
```
Comment (praneesha) [inline cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/cmd/PullCommand.java]:
```suggestion
        // Get the name of the module.
```

File: cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/cmd/Constants.java [hunk 9]
```diff
@@ -37,4 +37,11 @@
     static final String SEARCH_COMMAND = "search";
     static final String CLEAN_COMMAND = "clean";
     static final String UNINSTALL_COMMAND = "uninstall";
+
+    //module name format : <org-name>/<module-name> | <org-name>/<module-name>:<version>
```
Comment (praneesha) [inline cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/cmd/Constants.java]:
```suggestion
    //Module name format: <org-name>/<module-name> | <org-name>/<module-name>:<version>
```

File: cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/cmd/Constants.java [hunk 10]
```diff
@@ -37,4 +37,11 @@
...
+    //version format : 1, 1.*, 1.*.*
```
Comment (praneesha) [inline cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/cmd/Constants.java]:
```suggestion
    //Module version format: 1, 1.*, 1.*.*
```

File: cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/cmd/PullCommand.java [hunk 11]
```diff
@@ -81,28 +83,35 @@ public void execute() {
...
-            orgName = resourceName.substring(0, orgNameIndex);
-            if (orgName.equals("ballerina")) {
-                throw LauncherUtils.createLauncherException("`Ballerina` is the builtin organization and its modules"
-                                                                    + " are included in the runtime.");
-            }
-        } else {
-            throw LauncherUtils.createLauncherException("no module-name provided");
+        if (!validateModuleName(resourceName)) {
+            CommandUtil.printError(outStream,
+                    "invalid module name. Provide the module-name with the org-name ",
```
Comment (praneesha) [inline cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/cmd/PullCommand.java]:
```suggestion
                    "invalid module name. Provide the module name with the org name. ",
```

File: cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/cmd/PullCommand.java [hunk 12]
```diff
@@ -81,28 +83,35 @@ public void execute() {
...
+            return;
+        }
+
+        String[] moduleInfo = resourceName.split("/");
+        orgName = moduleInfo[0];
+        packageName = moduleInfo[1];
+        if (orgName.equals("ballerina")) {
+            throw LauncherUtils.createLauncherException("`ballerina` is the builtin organization and its modules"
+                    + " are included in the runtime.");
         }
```
Comment (warunalakshitha) [inline cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/cmd/PullCommand.java]: if (orgName.equals("ballerina") || orgName.equals("ballerinax") ) {
            throw LauncherUtils.createLauncherException("`ballerina' and 'ballerinax' are reserved " +
                "organization names used by Ballerina");
        }

File: cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/cmd/PullCommand.java [hunk 13]
```diff
@@ -81,28 +83,35 @@ public void execute() {
...
+                    "ballerina pull {<org-name>/<module-name> | <org-name>/<module-name>:<version>}",
+                    false);
+            Runtime.getRuntime().exit(1);
+            return;
+        }
+
+        String[] moduleInfo = resourceName.split("/");
+        orgName = moduleInfo[0];
+        packageName = moduleInfo[1];
+        if (orgName.equals("ballerina") || orgName.equals("ballerinax")) {
```
Comment (hevayo) [inline cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/cmd/PullCommand.java]: This should be removed we should not special case ballerina org

Comment (suganyasuven) [reply cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/cmd/PullCommand.java]: Removed.

**Pull request merged**